tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -272,7 +267,249 @@ constexpr void test_compiletime() {
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        std::span s{a, 10};
+        ranges::destroy(s);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        std::span s{a, 10};
+        ranges::destroy(s);
+        alloc.deallocate(a, 10);
+    }
+#endif // __cpp_lib_concepts
+    {
+        auto alloc = allocator<A<int>>{};
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            construct_at(a + i);
+        }
+        destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            construct_at(a + i);
+        }
+        destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+#ifdef __cpp_lib_concepts
+    {
+        auto alloc = std::allocator<A<int>>{};
+        A<int>* a  = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        ranges::destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+    {
+        auto alloc           = std::allocator<nontrivial_A<int>>{};
+        nontrivial_A<int>* a = alloc.allocate(10);
+        for (int i = 0; i < 10; ++i) {
+            ranges::construct_at(a + i);
+        }
+        ranges::destroy_n(a, 10);
+        alloc.deallocate(a, 10);
+    }
+#endif // __cpp_lib_concepts
+}
+static_assert((test_compiletime_destroy_variants(), true));
+
+template <class T, bool Construct = false, bool Destroy = false>
+struct Alloc {
+    using value_type = T;
+    using size_type  = std::size_t;
+
+    constexpr Alloc(int id_) noexcept : id(id_) {}
+
+    constexpr value_type* allocate(size_t n) {
+        assert(n == 10);
+        return allocator<T>{}.allocate(n);
+    }
+
+    constexpr void deallocate(value_type* ptr, size_t n) {
+        assert(n == 10);
+        allocator<T>{}.deallocate(ptr, n);
+    }
+
+    constexpr void construct(value_type* ptr, value_type n) requires(Construct) {
+        construct_at(ptr, n);
+    }
+
+    constexpr void destroy(value_type* ptr) requires(Destroy) {
+        destroy_at(ptr);
+    }
+
+    constexpr Alloc select_on_container_copy_construction() const noexcept {
+        return Alloc{id + 1};
+    }
+
+    constexpr size_type max_size() noexcept {
+        return std::numeric_limits<size_type>::max() / sizeof(value_type);
+    }
+
+    int id;
+};
+
+constexpr void test_compiletime_allocator_traits() {
+    {
+        storage_for<A<int>> a;
+        Alloc<A<int>> alloc{10};
+        assert(alloc.id == 10);
+
+        auto result = std::allocator_traits<Alloc<A<int>>>::allocate(alloc, 10);

```

Comment:
Observation, no change requested: For clarity, I recommend using distinct values for distinct purposes in test code. Here, `10` is being used for the allocator's ID, and `10` is also the number of `A<int>` elements being allocated. This makes it more difficult to glance at the source code and immediately recognize the meaning of a `10`.